src/shared/components/VideoTileBase/VideoTileCover.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState } from 'react'
+import { CSSTransition, SwitchTransition } from 'react-transition-group'
+import useResizeObserver from 'use-resize-observer'
+
+import {
+  SvgGlyphClose,
+  SvgGlyphDraft,
+  SvgGlyphHide,
+  SvgIllustrativeReupload,
+  SvgLargeEdit,
+  SvgLargeUploadFailed,
+  SvgOutlineVideo,
+} from '@/shared/icons'
+import { transitions } from '@/shared/theme'
+import { formatDurationShort } from '@/utils/time'
+
+import { PullUp } from './PullUp'
+import { Anchor } from './VideoTileBase.styles'
+import {
+  CoverContainer,
+  CoverDurationOverlay,
+  CoverHoverOverlay,
+  CoverIconWrapper,
+  CoverImage,
+  CoverImageContainer,
+  CoverNoImage,
+  CoverSkeletonLoader,
+  CoverThumbnailUploadFailed,
+  CoverTopLeftContainer,
+  CoverVideoPublishingStateOverlay,
+  CoverWrapper,
+  DELAYED_FADE_CLASSNAME,
+  PublishingStateText,
+  RemoveButton,
+  SkeletonHoverOverlay,
+  UploadProgressTransition,
+} from './VideoTileCover.style'
+
+import { VideoTileBaseMetaProps, VideoTilePublisherProps } from '.'
+import { Text } from '../Text'
+import { UploadProgressBar } from '../UploadProgressBar'
+
+type TileSize = 'small' | 'big' | undefined
+
+type VideoTileCoverProps = {
+  hasAssetUploadFailed?: boolean
+  videoHref?: string
+  setTileSize: React.Dispatch<React.SetStateAction<TileSize>>
+  tileSize: TileSize
+  duration?: number | null
+  thumbnailUrl?: string | null
+  isLoadingThumbnail?: boolean
+  isLoading?: boolean
+  title?: string | null
+  channelTitle?: string | null
+} & VideoTileBaseMetaProps &
+  VideoTilePublisherProps
+
+export const MIN_VIDEO_TILE_WIDTH = 250
+const SMALL_SIZE_WIDTH = 300
+
+export const VideoTileCover: React.FC<VideoTileCoverProps> = ({
+  hasAssetUploadFailed,
+  videoHref,
+  setTileSize,
+  tileSize,
+  onRemoveButtonClick,
+  onClick,
+  isLoading,
+  uploadStatus,
+  thumbnailUrl,
+  isLoadingThumbnail,
+  isDraft,
+  publisherMode,
+  videoPublishState,
+  duration,
+  isPullupDisabled,
+  onPullupClick,
+  removeButton,
+  title,
+  channelTitle,
+}) => {
+  const [failedLoadImage, setFailedLoadImage] = useState(false)
```

Comment:
I wonder whether we still need that? `useAsset` should catch all the errors with images so I guess with the new functionality you've added we could remove that?